tests/backend/test_GameManager.py

```diff
@@ -0,0 +1,34 @@
+from backend.GameManager import *
+import pytest
+
+
+@pytest.fixture
+def game_manager():
+    game_manager = GameManager(7, 9)
+    yield game_manager
+
+class TestGameManager:
+    def test_initial_turn(self, game_manager):
+        # Test if the initial turn is WHITE
+        assert game_manager.get_turn() == 'white'
+
+    def test_all_pieces_placed(self, game_manager):
+        # Test at initialization all pieces are not expected to be placed
+        assert not game_manager.placement_complete()
+
+    def test_piece_placement(self, game_manager):
+        # Test piece placement
+        res = game_manager.piece_placement(0, 0)
+        assert res == 1
+
+        game_manager.end_turn()
+        res = game_manager.piece_placement(0, 0)
+        assert res == "GameManagerError -- invalid piece placement position"
```

Comment:
From the business logic perspective this should be an exception not a string,

Reply:
I agree the return type should be exception. We should change that in that function.